server/public/model/post.go

```diff
@@ -657,6 +664,152 @@ func (o *Post) GetProp(key string) any {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_bot prop must be a string"))
+		} else if fromBot != "true" {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_bot prop must be \"true\""))
+		}
+	}
+	if props[PostPropsFromOAuthApp] != nil {
+		if fromOAuthApp, ok := props[PostPropsFromOAuthApp].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_oauth_app prop must be a string"))
+		} else if fromOAuthApp != "true" {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_oauth_app prop must be \"true\""))
+		}
+	}
+	if props[PostPropsFromPlugin] != nil {
+		if fromPlugin, ok := props[PostPropsFromPlugin].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_plugin prop must be a string"))
+		} else if fromPlugin != "true" {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("from_plugin prop must be \"true\""))
+		}
+	}
+	if props[PostPropsUnsafeLinks] != nil {
+		if unsafeLinks, ok := props[PostPropsUnsafeLinks].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("unsafe_links prop must be a string"))
+		} else if unsafeLinks != "true" {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("unsafe_links prop must be \"true\""))
+		}
+	}
+	if props[PostPropsWebhookDisplayName] != nil {
+		if _, ok := props[PostPropsWebhookDisplayName].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("webhook_display_name prop must be a string"))
+		}
+	}
+
+	if props[PostPropsMentionHighlightDisabled] != nil {
+		if _, ok := props[PostPropsMentionHighlightDisabled].(bool); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("mention_highlight_disabled prop must be a boolean"))
+		}
+	}
+	if props[PostPropsGroupHighlightDisabled] != nil {
+		if _, ok := props[PostPropsGroupHighlightDisabled].(bool); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("disable_group_highlight prop must be a boolean"))
+		}
+	}
+
+	if props[PostPropsPreviewedPost] != nil {
+		if previewedPostID, ok := props[PostPropsPreviewedPost].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("previewed_post prop must be a string"))
+		} else if !IsValidId(previewedPostID) {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("previewed_post prop must be a valid post ID"))
+		}
+	}
+
+	if props[PostPropsForceNotification] != nil {
+		if _, ok := props[PostPropsForceNotification].(bool); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("force_notification prop must be a boolean"))
+		}
+	}
+
+	if props[PostPropsRequestedAck] != nil {
+		if _, ok := props[PostPropsRequestedAck].(bool); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("requested_ack prop must be a boolean"))
+		}
+	}
+	if props[PostPropsPersistentNotifications] != nil {
+		if _, ok := props[PostPropsPersistentNotifications].(bool); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("persistent_notifications prop must be a boolean"))
+		}
+	}
```

Comment:
Sanity check: In many props we check for the "true" string, but these look for booleans. Is that right?